File: back/src/main/java/fr/mds/debazar/core/model/Offer.java
package fr.mds.debazar.core.model;

import java.sql.Date;
import java.sql.Timestamp;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import fr.mds.debazar.authentication.model.User;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name= "OFFER")
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "game", referencedColumnName = "id")
    private Game game = new Game();
    @ManyToOne
    @JoinColumn(name = "seller")
    private User seller;
    @ManyToOne
    @JoinColumn(name = "state")
    private State state;

    private String detail;
    private float price;

    private Timestamp date;

    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }

    public Game getGame() {
        return game;
    }
    public void setGame(Game game) {
        this.game = game;
    }

    public User getSeller() {
        return seller;
    }
    public void setSeller(User seller) {
        this.seller = seller;
    }

    public State getState() {
        return state;
    }
    public void setState(State state) {
        this.state = state;
    }

    public String getDetail() {
        return detail;
    }
    public void setDetail(String detail) {
        this.detail = detail;
    }

    public float getPrice() {
        return price;
    }
    public void setPrice(float price) {
        this.price = price;
    }

    public Timestamp getDate() {
        return date;
    }
    public void setDate(Timestamp	 date) {
        this.date = date;
    }

}

File: front/src/app/global/models/Offer.ts
import { State } from './State';
import User from './User';
import Game from './Game';

export class Offer {

    // Attributes
    private _id: number;
    private _game: Game;
    private _seller: User;
    private _state: State;
    private _detail: string;
    private _price: number;

    // Constructor
    constructor(id: number, game: Game, seller: User, state: State, detail: string, price: number) {
        this._id = id;
        this._game = game;
        this._seller = seller;
        this._state = state;
        this._detail = detail;
        this._price = price;
    }

    // Getters / Setters
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get game(): Game {
        return this._game;
    }
    public set game(value: Game) {
        this._game = value;
    }
    public get seller(): User {
        return this._seller;
    }
    public set seller(value: User) {
        this._seller = value;
    }
    public get state(): State {
        return this._state;
    }
    public set state(value: State) {
        this._state = value;
    }
    public get detail(): string {
        return this._detail;
    }
    public set detail(value: string) {
        this._detail = value;
    }
    public get price(): number {
        return this._price;
    }
    public set price(value: number) {
        this._price = value;
    }
}